packages/core/src/controllers/verify.ts

```diff
@@ -47,21 +68,35 @@ export class Verify extends IVerify {
 
   public resolve: IVerify["resolve"] = async (params) => {
     if (this.isDevEnv) return "";
+    const mainUrl = params?.verifyUrl || VERIFY_SERVER;
+    let result = "";
+    try {
+      result = await this.fetch(params.attestationId, mainUrl);
+    } catch (error) {
+      this.logger.warn(
+        `failed to resolve attestation: ${params.attestationId} from url: ${mainUrl}`,
+      );
+      this.logger.warn(error);
+      result = await this.fetch(params.attestationId, VERIFY_FALLBACK_SERVER);
+    }
+    return result;
+  };
 
-    this.logger.info(`resolving attestation: ${params.attestationId}`);
+  get context(): string {
+    return getLoggerContext(this.logger);
+  }
+
+  private fetch = async (attestationId: string, url: string) => {
```

Comment:
Nit: not a big deal but feel this would be more accurate and avoids kind of shadowing the `fetch` keyword (even though it's a class method here)

```suggestion
  private fetchAttestation = async (attestationId: string, url: string) => {
```